ddtrace/contrib/kafka/patch.py

```diff
@@ -131,10 +133,7 @@ def traced_produce(func, instance, args, kwargs):
     partition = kwargs.get("partition", -1)
     if config._data_streams_enabled:
         # inject data streams context
-        headers = kwargs.get("headers", {})
-        pathway = pin.tracer.data_streams_processor.set_checkpoint(["direction:out", "topic:" + topic, "type:kafka"])
-        headers[PROPAGATION_KEY] = pathway.encode()
-        kwargs["headers"] = headers
+        core.dispatch("kafka.produce.start", [instance, args, kwargs])
```

Comment:
do we have a pattern for what is placed in the dispatch vs what is attached to the context (via `core.set_item()`) ?

Reply:
As an extension:

I'm not sure where the differentiation is between core and product for extracting information.  I'm currently on the side of putting the extraction code into the product, which means passing/attaching the already-available information and then deriving values in the product.  However, this could mean each product does its own derivation.  

In the case of "kafka_topic" I'd pass "message" to the product code and evaluate `message.product` there.

Reply:
> do we have a pattern for what is placed in the dispatch vs what is attached to the context

No. Hopefully one will arise as we transition more integrations to the core API.

> I'm not sure where the differentiation is between core and product for extracting information.

In my opinion, parsing the message once in the integration code (this file) and passing kafka_topic to `dispatch` is the simpler separation of concerns. Aside from not forcing the Tracing product to know how to parse a Kafka message, it also means that the possibly-expensive parsing operation only happens once rather than per-product.